<launch>
  <arg name="robot_name" default="turtlebot1"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="yaw" default="0.0"/>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="move_forward_only" default="true"/>

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model $(arg robot_name)
      -x $(arg x_pos) -y $(arg y_pos) -Y $(arg yaw) -param /robot_description" />

  <!-- static map publisher -->
  <node pkg="tf" type="static_transform_publisher" name="map_odom_publisher" args="0 0 0 0 0 0 1 map $(arg robot_name)_tf/odom 100" />

  <include file="$(find turtlebot3_bringup)/launch/includes/description.launch.xml">
    <arg name="model" value="$(arg model)" />
  </include>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="50.0" />
  </node>

  <!-- move_base -->
  <include file="$(find navlab_turtlebot_base)/launch/move_base_multi.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="robot_name" value="$(arg robot_name)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    <arg name="odom_topic" value="$(arg robot_name)_tf/odom"/>
  </include>

</launch>
